src/librustc/traits/error_reporting.rs

```diff
@@ -202,17 +202,19 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
                     obligation.cause.span,
                     infer::LateBoundRegionConversionTime::HigherRankedType,
                     data);
-                let normalized = super::normalize_projection_type(
+                let mut obligations = vec![];
```

Comment:
If this specifically is hot, it'd be interesting to see if we could make this an `Option` argument since `obligations` appear to be unused in this case.

Reply:
This particular callsite is not hot. The hot callsite is the one within `<TypeFolder for AssociatedTypeNormalizer>::fold_ty` in `librustc/traits/project.rs`, which passes an existing vector.